Added docket tag note to security section (#96725)
grafana/grafana: docs/sources/upgrade-guide/when-to-upgrade/index.md

We've enhanced our naming convention for security release versions to make it easier to clearly identify our security releases from our standard patching releases.

In the past, critical vulnerabilities triggered unscheduled releases that incremented the patch version (e.g., 10.3.0 to 10.3.1). However, we found that the naming convention for these releases didn't clearly communicate the nature of the update. For example, if there was a version change from 11.3.0 to 11.3.1, there was no indication whether it was a security fix, a bug fix, or a minor feature update. This lack of clarity led to confusion about the urgency and nature of the update. <br>

> {{< admonition type="note" >}}

Reviewer: ```suggestion

{{< admonition type="note" >}}
```

We've enhanced our naming convention for security release versions to make it easier to clearly identify our security releases from our standard patching releases.

In the past, critical vulnerabilities triggered unscheduled releases that incremented the patch version (e.g., 10.3.0 to 10.3.1). However, we found that the naming convention for these releases didn't clearly communicate the nature of the update. For example, if there was a version change from 11.3.0 to 11.3.1, there was no indication whether it was a security fix, a bug fix, or a minor feature update. This lack of clarity led to confusion about the urgency and nature of the update. <br>

{{< admonition type="note" >}}

> Docker does not allow the plus sign (`+`) sign in image tag names. A plus sign (`+`) will be a rendered as dash (`-`) in the docker tag.

Reviewer: ```suggestion
Docker does not allow the plus sign (`+`) in image tag names. A plus sign (`+`) will be a rendered as a dash (`-`) in the docker tag.
```

We've enhanced our naming convention for security release versions to make it easier to clearly identify our security releases from our standard patching releases.

In the past, critical vulnerabilities triggered unscheduled releases that incremented the patch version (e.g., 10.3.0 to 10.3.1). However, we found that the naming convention for these releases didn't clearly communicate the nature of the update. For example, if there was a version change from 11.3.0 to 11.3.1, there was no indication whether it was a security fix, a bug fix, or a minor feature update. This lack of clarity led to confusion about the urgency and nature of the update. <br>

> *A note about docker tags: docker does not allow the + sign in image tag names. As a result the + will be a - in the docker tag.* 

Reviewer: ```suggestion
{{< admonition type="note" >}}
Docker does not allow the plus sign (`+`) sign in image tag names. A plus sign (`+`) will be a rendered as dash (`-`) in the docker tag.
{{< /admonition >}}
```
